contracts/governance/src/ZeroExVotes.sol

```diff
@@ -0,0 +1,359 @@
+
+                // Remove the entire source delegator's sqrt balance from delegatee's voting power.
+                // `srcDelegateBalance` is value _after_ transfer so add the amount that was transferred.
+                if (pos > 0)
+                    oldCkptSrcDelegate.quadraticVotes -= SafeCast.toUint48(Math.sqrt(srcDelegateBalance + amount));
+
+                uint256 newLinearBalance = oldCkptSrcDelegate.votes - amount;
+                uint256 newQuadraticBalance = oldCkptSrcDelegate.quadraticVotes + Math.sqrt(srcDelegateBalance);
+
+                _writeCheckpoint(_checkpoints[srcDelegatee], newLinearBalance, newQuadraticBalance);
+
+                emit DelegateVotesChanged(srcDelegatee, oldCkptSrcDelegate.votes, newLinearBalance);
+
+                emit DelegateQuadraticVotesChanged(
+                    srcDelegatee,
+                    oldCkptSrcDelegate.quadraticVotes,
+                    newQuadraticBalance
+                );
+            }
+
+            if (dstDelegatee != address(0)) {
+                uint256 pos = _checkpoints[dstDelegatee].length;
+                Checkpoint memory oldCkptDstDelegate = pos == 0
+                    ? Checkpoint(0, 0, 0)
+                    : _unsafeAccess(_checkpoints[dstDelegatee], pos - 1);
+
+                // Remove the entire destination delegator's sqrt balance from delegatee's voting power.
+                // `dstDelegateBalance` is value _after_ transfer so remove the amount that was transferred.
+                if (pos > 0)
+                    oldCkptDstDelegate.quadraticVotes -= SafeCast.toUint48(Math.sqrt(dstDelegateBalance - amount));
+
+                uint256 newLinearBalance = oldCkptDstDelegate.votes + amount;
+                uint256 newQuadraticBalance = oldCkptDstDelegate.quadraticVotes + Math.sqrt(dstDelegateBalance);
+
+                _writeCheckpoint(_checkpoints[dstDelegatee], newLinearBalance, newQuadraticBalance);
+
+                emit DelegateVotesChanged(dstDelegatee, oldCkptDstDelegate.votes, newLinearBalance);
+
+                emit DelegateQuadraticVotesChanged(
+                    dstDelegatee,
+                    oldCkptDstDelegate.quadraticVotes,
+                    newQuadraticBalance
+                );
+            }
+        }
+    }
+
+    /**
+     * @inheritdoc IZeroExVotes
+     */
+    function writeCheckpointTotalSupplyMint(uint256 amount, uint256 accountBalance) public override onlyToken {
+        uint256 pos = _totalSupplyCheckpoints.length;
+        Checkpoint memory oldCkptTotalSuply = pos == 0
+            ? Checkpoint(0, 0, 0)
+            : _unsafeAccess(_totalSupplyCheckpoints, pos - 1);
+
+        // Remove the account sqrt balance from total quadratic supply.
+        // `accountBalance` is value _after_ minting
+        if (pos > 0) oldCkptTotalSuply.quadraticVotes -= SafeCast.toUint48(Math.sqrt(accountBalance - amount));
+
+        uint256 newLinearBalance = oldCkptTotalSuply.votes + amount;
```

Comment:
If I understand it correctly. the total supply quadratic votes is tracked as `sum(sqrt(user_balance))` which is higher than the `sqrt(sum(user_balance))`

in fact to reach the treasury threshold, I think it will need less than the 250K ZRX

example:
4 users with 25K each, have ~ 158 voting power each which combined will be 632 which is way bigger than the 500 = sqrt(250K) threshold and I think this gets worse the more users are involved